src/exec.c

```diff
@@ -726,6 +727,49 @@ struct hyper_exec *hyper_find_exec_by_seq(struct hyper_pod *pod, uint64_t seq)
 	return NULL;
 }
 
+static int hyper_kill_container_processes(struct hyper_container *c) {
+	struct stat st;
+	int pid;
+	DIR *dp;
+	struct dirent *de;
+
+	if (fstat(c->ns, &st) < 0) {
+		perror("fail to stat mnt ns");
+		return -1;
+	}
+
+	fprintf(stdout, "container init process %d\n", c->exec.pid);
+	dp = opendir("/proc");
+	if (dp == NULL)
+		return -1;
+
+	while ((de = readdir(dp)) && de != NULL) {
+		char mntns[512];
+		struct stat st1;
+
+		if (!isdigit(de->d_name[0]))
+			continue;
+		pid = atoi(de->d_name);
+		if (pid == 1 || pid == c->exec.pid)
+			continue;
+
+		sprintf(mntns, "/proc/%d/ns/mnt", pid);
+		if (stat(mntns, &st1) < 0) {
+			perror("fail to stat process mnt ns");
+			continue;
+		}
+
+		if (st.st_ino != st1.st_ino)
+		       continue;
+
+		fprintf(stdout, "kill process of container %d\n", pid);
+		kill(pid, SIGKILL);
+	}
+
+	closedir(dp);
```

Comment:
a outer loop is needed.
```c
for (;;) {
   L742-769
   if there is no process was killed {
      break;
   }
}
```
new processes can be being created when the function is called.